lib/client/ipam.go

```diff
@@ -486,7 +486,7 @@ func (c ipams) assignFromExistingBlock(
 
 		// Update the block using CAS by passing back the original
 		// KVPair.
-		obj.Value = &b.AllocationBlock
+		obj.Value = b.AllocationBlock
```

Comment:
You should undo this change.  It's currently expected that KVPair.Value fields are pointers since https://github.com/projectcalico/libcalico-go/pull/181